### PR TITLE
[CUMULUS-2903]: Bump Node version to 14.19.1

### PR DESCRIPTION
**Pull request by npauzenga**
This updates Node from 12.18 to 14.19.1, the latest version of Node 14. See https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V14.md#14.19.1

We're using Node 14 instead of a more recent version because 14 is the latest available AWS runtime.

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Ad-hoc testing - Deploy changes and test manually
- [x] Integration tests

### CODE REVIEW

File: bamboo/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ## Dockerfile to create integration/unit test environment
-FROM node:12.22-buster
+FROM node:14.19.1-buster
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply bamboo/Dockerfile:2]: CI won't catch issues with this container build as it's going to be using the old cached version, as well.    Did you test it locally by building and running the units against the updated container?

File: docs/workflows/lambda.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ AWS provides great getting started guide for building Lambdas in the [developer
 
 Cumulus currently supports the following environments for Cumulus Message Adapter enabled functions:
 
-* [Node.js 12.18](https://docs.aws.amazon.com/lambda/latest/dg/programming-model.html)
+* [Node.js 14.19.1](https://docs.aws.amazon.com/lambda/latest/dg/programming-model.html)
```
Comment (Jkovarik) [inline docs/workflows/lambda.md:13]: Technically we're still supporting 12.18 right?   The CMA requirements aren't being updated as part of this?

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
+### Notable Changes
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply CHANGELOG.md:13]: I think it *is* breaking if our npm packages are no longer importable in node 12?   I realize that's likely *not* the case, for now.

**Review by Jkovarik — CHANGES_REQUESTED** [review]
This all looks great! Just a couple of things need addressed in the comments and:

- Will this pass in CI given we can't publish the docker image given the way that works?  I think we need to set up a build environment to validate this will work on merge before merging it.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply CHANGELOG.md:13]: Also we need a user note to grab the newest ecs-task image for our tasks.

File: .vscode.example/launch.json [hunk 4]
```diff
@@ -19,7 +19,7 @@
       ],
       "port": 9229,
       "outputCapture": "std",
-      "runtimeVersion": "12.18.2",
+      "runtimeVersion": "14.19.1",
```
[the remark this answers is not in the document]
Comment (npauzenga) [reply .vscode.example/launch.json:22]: Yeah, I'm pretty confident everything here works as it should. I verified that my .vscode/launch.json was using the 14.19.1 runtime, did a fresh npm install and boostrap, and the debugger runs fine. If you can think of anything else to test here I'm game.

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
   - Updates migration script with new `migrateAndOverwrite` and
     `migrateOnlyFiles` options.
 
+### Notable Changes
+
+- **CUMULUS-2903**
+  - The minimum supported version for all Cumulus packages is now Node 14.19.1
```
[the remark this answers is not in the document]
Comment (npauzenga) [reply CHANGELOG.md]: How about this -- https://github.com/nasa/cumulus/pull/2917/commits/e230ef7c6ee2fa13d1dcaecbb43c188a42117223

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (npauzenga) [reply CHANGELOG.md]: Cool, made the update here. The wording here is mostly stolen from the last Node update: https://github.com/nasa/cumulus/pull/2917/commits/2e54fb808038f7c977aa327cd3d26a86b759abb3

Also opened a PR for updating ecs-task: https://github.com/nasa/cumulus-ecs-task/pull/48

I don't know much about the cumulus-ecs-task repo so I could maybe use a chat about it. Particularly how to validate and test changes.

[the remark this answers is not in the document]
Comment (Jkovarik) [reply CHANGELOG.md]: Yeah, I'd be happy to do that.   Let's pair on this tomorrow.

File: docs/workflows/lambda.md (hunk 2 above)
Comment (Jkovarik) [reply docs/workflows/lambda.md]: Hmm, no, I talked myself out of it.   Or at least it should be > 12.18?     🤔

File: CHANGELOG.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply CHANGELOG.md]: Thanks for the change, that looks good.

File: bamboo/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (npauzenga) [reply bamboo/Dockerfile:2]: Resolving this comment as we're now testing against a new Node 14 build plan in Bamboo

File: docs/workflows/lambda.md (hunk 2 above)
Comment (npauzenga) [reply docs/workflows/lambda.md]: 🤔  so you're suggesting saying that the CMA supports Node versions > 12.18?

Comment (Jkovarik) [reply docs/workflows/lambda.md]: What I mean is: 

If we update CMA *only* to v14 and release a new version, and assuming that for whatever reason a v14 CMA release build cannot be imported by a v12 lambda, Core will work with *either* CMA version as functionally we haven't made a change to the CMA or the message, so we still support 12.18 tasks.     

As always, this dependency tree is clear as mud.

**Comment by npauzenga** [comment]
Ok, I think I have the CI situation tested. The build running on this branch takes place on a new build plan that uses a new container built from this branch: `maven.earthdata.nasa.gov/cumulus:142`. That container is a Node 14 runtime.

The plan configuration: https://ci.earthdata.nasa.gov/build/admin/edit/editBuildDocker.action?buildKey=CUM-CCN1-CUBPR

Steps to merge the Node upgrade:
1. Merge this branch to master
2. Manually run the Cumulus Image Publisher job against master to build and deploy a Node 14 container on the `latest` image.
3. Profit. Node 12 builds _should_ work on a Node 14 container so we _should not_ run into issues where existing branches fail against the new image. If there are issues we will need to merge master into any impacted feature branch.

File: example/spec/parallel/pythonReferenceTests/PythonReferenceSpec.js [hunk 6]
```diff
@@ -36,6 +37,10 @@ describe('The Python Reference workflow', () => {
     activityStep = new ActivityStep();
   });
 
+  afterAll(async () => {
+    await deleteExecution({ prefix: config.stackName, executionArn: workflowExecution.executionArn });
+  });
```
Comment (npauzenga) [inline example/spec/parallel/pythonReferenceTests/PythonReferenceSpec.js:42]: Added this bit of cleanup. I was having integration test issues with my stack and noticed that this test wasn't cleaning up (verified that executions remain in Elastic after the test completes). This wasn't actually the cause of my problem but it seemed good to leave this in anyway.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply CHANGELOG.md]: I see the related PR, I'm assuming we have to release that before a release?     I know node 12 *should* work with the update, so perhaps that's a moot point.

File: docs/workflows/lambda.md (hunk 2 above)
Comment (Jkovarik) [reply docs/workflows/lambda.md]: I think we still need to note that we *support* v14 though, right?    So either both, or ">" or something like that.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jkovarik) [reply CHANGELOG.md]: I'm thinking we need to update example/cumulus-tf/variables.tf with a new default and push the results of https://github.com/nasa/cumulus-ecs-task/pull/48 to validate both.

[the remark this answers is not in the document]
Comment (npauzenga) [reply CHANGELOG.md]: So the steps would be:

1. Merge and publish https://github.com/nasa/cumulus-ecs-task/pull/48 (it looks like the publish to DockerHub is done outside of Circle CI?)
2. Update `ecs_task_image_version` in `example` to the new image version
3. Deploy Core
4. Verify by... running something like data-migration1 or another function that runs async in an ECS container?

File: docs/workflows/lambda.md (hunk 2 above)
Comment (npauzenga) [reply docs/workflows/lambda.md]: How about https://github.com/nasa/cumulus/pull/2917/commits/b8bfc02d31530b8aaaf72f6fcd4dfb670273e1b1?

Wanted to make it clear that the max we support (due to AWS constraints) is 14.

**Review by Jkovarik — APPROVED** [review]
Looks good.  Approving, but needs int tests to pass on the new ecs version before merge.

**Pull request merged**